src/other/utils.jl

```diff
@@ -221,16 +221,23 @@ end
 Equivalent to `Threads.@spawn` if `threads === true`,
 otherwise run `expr` and return a `Task` that returns its value.
 """
-macro spawn_or_run_task(threads, expr)
-    letargs = Base._lift_one_interp!(expr)
+macro spawn_or_run_task(threads, ex)
+    letargs = Base._lift_one_interp!(ex)
 
-    thunk = esc(:(()->($expr)))
+    thunk = :(()->($(esc(ex))))
+    @static if VERSION >= v"1.10.0-DEV"
+        Base.replace_linenums!(thunk, __source__)
+    end
     var = esc(Base.sync_varname)
+    spawn_set_thrpool = VERSION >= v"1.9.0" ?
+        :(Base.Threads._spawn_set_thrpool(task, :default)) :
+        :()
     quote
         let $(letargs...)
             if $(esc(threads))
                 local task = Task($thunk)
                 task.sticky = false
+                $(spawn_set_thrpool)
```

Comment:
Just to be 100% sure. Is the macro hygiene here correct? I.e. the `task` variable will be left untouched in the interpolated expression and refer to the local `task` from line 238?.

Reply:
AFAIK local variables names are only replaced with `gensym` as a final step once the macro has returned an expression, so it doesn't make a difference whether we build the expression in multiple steps. What makes me confident that it's the case is that tests pass. ;-)